src/opm/parser/eclipse/EclipseState/Schedule/Well/WellTestState.cpp

```diff
@@ -31,11 +31,19 @@ namespace Opm {
         WTestWell* well_ptr = getWell(well_name, reason);
 
         if (well_ptr) {
+            if (well_ptr->closed) {
+                throw std::runtime_error( " Well " + well_name + " is closed with reason "
+                                        + WellTestConfig::reasonToString(reason)
+                                        + ", we ware trying to close it again with same reason!");
+            }
             // the well exists already, we just update it with action of closing
             well_ptr->closed = true;
             well_ptr->last_test = sim_time;
-        } else
-            this->wells.push_back({well_name, reason, true, sim_time, 0});
+        } else {
+            // by default, we use -1 if there is no WTEST request for this well
+            // it will be updated when checking for WellTestConfig
+            this->wells.push_back({well_name, reason, true, sim_time, 0, -1});
```

Comment:
Should be possible to pass the `WTestWell` structure to this function, and thereby get access to the correct report step from the start, and hence completely avoiding the temporary illegal -1 value.

Reply:
> Should be possible to pass the WTestWell structure to this function,

yes. it makes the usage of the function a little more complicated in the simulator side. If you insist this way, I will do it. 

> and hence completely avoiding the temporary illegal -1 value.

Nope. For example, we only have `WECON`, without `WTEST`.  We will need to shut/close well due to economic reason, while we do not want to do `WTEST`, then there is no WTEST associated with the shut/closed well.